storage/src/metadata/storage.rs

```diff
@@ -36,55 +38,66 @@ impl<E: Clock + Storage + Metrics, K: Array> Metadata<E, K> {
 
         // Set checksums
         let mut left_version = 0;
-        let mut left_data = BTreeMap::new();
-        if let Some((version, data)) = left_result {
+        let mut left_map = BTreeMap::new();
+        let mut left_data = Vec::new();
+        if let Some((version, map, data)) = left_result {
             left_version = version;
+            left_map = map;
             left_data = data;
         }
         let mut right_version = 0;
-        let mut right_data = BTreeMap::new();
-        if let Some((version, data)) = right_result {
+        let mut right_map = BTreeMap::new();
+        let mut right_data = Vec::new();
+        if let Some((version, map, data)) = right_result {
             right_version = version;
+            right_map = map;
             right_data = data;
         }
 
         // Choose latest blob
-        let mut data = left_data;
+        let mut map = left_map;
         let mut cursor = 0;
         if right_version > left_version {
             cursor = 1;
-            data = right_data;
+            map = right_map;
         }
 
         // Create metrics
         let syncs = Counter::default();
         let keys = Gauge::default();
+        let skipped = Counter::default();
         context.register("syncs", "number of syncs of data to disk", syncs.clone());
         context.register("keys", "number of tracked keys", keys.clone());
+        context.register(
+            "skipped",
+            "total bytes not written to disk",
+            skipped.clone(),
+        );
 
         // Return metadata
-        keys.set(data.len() as i64);
+        keys.set(map.len() as i64);
         Ok(Self {
             context,
 
-            data,
+            map,
             cursor,
             partition: cfg.partition,
             blobs: [
-                (left_blob, left_len, left_version),
-                (right_blob, right_len, right_version),
+                (left_blob, left_version, left_data),
+                (right_blob, right_version, right_data),
             ],
 
             syncs,
             keys,
+            skipped,
         })
     }
 
     async fn load(
         index: usize,
         blob: &E::Blob,
         len: u64,
-    ) -> Result<Option<(u64, BTreeMap<K, Vec<u8>>)>, Error<K>> {
+    ) -> Result<Option<(u64, BTreeMap<K, Vec<u8>>, Vec<u8>)>, Error<K>> {
```

Comment:
This return type is kind of complicated. Could we document what the fields mean?